lib/reline/ansi.rb

```diff
@@ -60,8 +61,36 @@ def self.set_default_key_bindings(config)
       config.add_default_key_binding_by_keymap(:emacs, key, func)
     end
   end
+  
+  @@input = STDIN
+  def self.input=(val)
+    @@input = val
+  end
+
+  @@output = STDOUT
+  def self.output=(val)
+    @@output = val
+  end
 
   def self.set_default_key_bindings_terminfo(config)
+    
+    # put terminal in tx mode so that
+    # keys match terminfo's bindings
+    begin
+      @@output.write Reline::Terminfo.tigetstr('smkx')
+    rescue Reline::Terminfo::TerminfoError
+      # capname is undefined
+    end
+    
+    # return terminal to rx mode
+    at_exit do
```

Comment:
I think we should send the "smkx" and "rmkx" in `prep` and `deprep` rather than doing it here in set_default_key_bindings_terminfo and having to rely on an `at_exit`, which can be problematic.